wurst/systems/spells/gatherer/Radars.wurst

```diff
@@ -305,20 +348,39 @@ function onCast(int itemId)
         range += MAP_MAGIC_RANGE_BONUS
 
     forItemsInMap() i ->
-        if i.getPos().distanceTo(casterPos) < range and i.getTypeId() == itemId and not areHostilesNearItem(owner, i)
-            //Vision
-            var vis = createVision(owner, i.getPos(), 128, true)
-            vis.start()
-            doAfter(REVEAL_DURATION) ->
-                vis.destr()
+        if i.getPos().distanceTo(casterPos) < range and i.getTypeId() == itemId and i.isPickupable()
+            //Fog Dummy
+            if not item2radarDummy.has(i)
+                let dummy = createUnit(players[PLAYER_NEUTRAL_PASSIVE], dummyDictionary.get(i.getTypeId()), i.getPos(), (270.0).asAngleDegrees())
+                dummy.setVertexColor(255, 255, 255, 0)
+                radarDummyDeath.registerDeathEvent(i)
+                item2radarDummy.put(i, dummy)
+                dummy.addAbility('Amrf')
+                dummy.removeAbility('Amrf')
+                let team = Tribe.ofPlayer(owner).getMembers()
+                for teammate in team
+                    if teammate == GetLocalPlayer()
+                        dummy.setVertexColor(255, 255, 255, 255)
+            else
+                let team = Tribe.ofPlayer(owner).getMembers()
```

Comment:
You're repeating the coloring logic in both branches, so just move that outside. Insert a new object and then retrieve it. The overhead of doing a put and a get isn't worth the code I think, plus Wurst

Reply:
No put and get are being done in any one branch. But otherwise I made the reveal code into a function.